gpflow/models/model.py

```diff
@@ -43,41 +43,25 @@ def log_prior_density(self) -> tf.Tensor:
         else:
             return tf.convert_to_tensor(0.0, dtype=default_float())
 
-    def log_posterior_density(self, data: Optional[Data] = None) -> tf.Tensor:
+    def log_posterior_density(self, *args, **kwargs) -> tf.Tensor:
         """
         This may be the posterior with respect to the hyperparameters (e.g. for
         GPR) or the posterior with respect to the function (e.g. for GPMC and
         SGPMC). It assumes that maximum_likelihood_objective() is defined
         sensibly.
         """
-        return self._call_maximum_likelihood_objective(data) + self.log_prior_density()
+        return self.maximum_likelihood_objective(*args, **kwargs) + self.log_prior_density()
 
-    def training_loss(self, data: Optional[Data] = None) -> tf.Tensor:
+    def _training_loss(self, *args, **kwargs) -> tf.Tensor:
```

Comment:
Docstring discussing data input?
`Data will be an accepted input for BayesianModelWithoutData and will fail on BayesianModelWithData as in the latter case the model already has data.`

Reply:
I will add that and generally clean it up if this draft is a structure that would make everyone happy